dronegen/tag.go

```diff
@@ -0,0 +1,373 @@
+		commands = append(commands,
+			fmt.Sprintf(`aws s3 cp s3://$AWS_S3_BUCKET/teleport/$${S3_PATH}teleport-v$${VERSION}-%s-bin.tar.gz /go/artifacts/`, artifactType),
+		)
+	}
+	if artifactEnterprise {
+		commands = append(commands,
+			fmt.Sprintf(`aws s3 cp s3://$AWS_S3_BUCKET/teleport/$${S3_PATH}teleport-ent-v$${VERSION}-%s-bin.tar.gz /go/artifacts/`, artifactType),
+		)
+	}
+	return commands
+}
+
+// tagCopyArtifactCommands generates a set of commands to find and copy built package artifacts as part of a tag build
+func tagCopyArtifactCommands(params tagBuildType, packageType string) []string {
+	commands := []string{
+		`cd /go/src/github.com/gravitational/teleport`,
+	}
+	if !params.fips {
+		commands = append(commands, fmt.Sprintf(`find build -maxdepth 1 -iname "teleport*.%s*" -print -exec cp {} /go/artifacts \;`, packageType))
+	}
+	commands = append(commands, fmt.Sprintf(`find e/build -maxdepth 1 -iname "teleport*.%s*" -print -exec cp {} /go/artifacts \;`, packageType))
+	return commands
+}
+
+// tagPackagePipeline generates a tag package pipeline for a given combination of os/arch/FIPS
+func tagPackagePipeline(packageType string, params tagBuildType) pipeline {
+	if packageType == "" {
+		panic("packageType must be set")
+	}
+	if params.os == "" {
+		panic("params.os must be set")
+	}
+	if params.arch == "" {
+		panic("params.arch must be set")
+	}
+
+	environment := map[string]value{
+		"ARCH":             value{raw: params.arch},
+		"TMPDIR":           value{raw: "/go"},
+		"ENT_TARBALL_PATH": value{raw: "/go/artifacts"},
+	}
+
+	dependentPipeline := fmt.Sprintf("build-%s-%s", params.os, params.arch)
+	packageBuildCommands := []string{
+		`apk add --no-cache bash curl gzip make tar`,
+		`cd /go/src/github.com/gravitational/teleport`,
+		`export VERSION=$(cat /go/.version.txt)`,
+	}
+
+	makeCommand := fmt.Sprintf("make %s", packageType)
+	if params.fips {
+		dependentPipeline += "-fips"
+		environment["FIPS"] = value{raw: "yes"}
+		environment["RUNTIME"] = value{raw: "fips"}
+		makeCommand = fmt.Sprintf("make -C e %s", packageType)
+	} else {
+		environment["OSS_TARBALL_PATH"] = value{raw: "/go/artifacts"}
+	}
+
+	if packageType == rpmPackage {
+		environment["GNUPG_DIR"] = value{raw: "/tmpfs/gnupg"}
+		environment["GPG_RPM_SIGNING_ARCHIVE"] = value{fromSecret: "GPG_RPM_SIGNING_ARCHIVE"}
+		packageBuildCommands = append(packageBuildCommands,
+			`mkdir -m0700 $GNUPG_DIR`,
+			`echo "$GPG_RPM_SIGNING_ARCHIVE" | base64 -d | tar -xzf - -C $GNUPG_DIR`,
+			`chown -R root:root $GNUPG_DIR`,
+			makeCommand,
+			`rm -rf $GNUPG_DIR`,
+		)
+	} else if packageType == debPackage {
+		packageBuildCommands = append(packageBuildCommands,
+			makeCommand,
+		)
+	}
+
+	pipelineName := fmt.Sprintf("%s-%s", dependentPipeline, packageType)
+
+	p := newKubePipeline(pipelineName)
+	p.Trigger = triggerTag
+	p.DependsOn = []string{dependentPipeline}
+	p.Workspace = workspace{Path: "/go"}
+	p.Volumes = []volume{
+		{Name: "dockersock", Temp: &volumeTemp{}},
```

Comment:
nit: since this comes up on multiple occasions, this could potentially be factored out as `volumeDocker`